src/utils/auto-box-collider.js

```diff
@@ -1,13 +1,70 @@
+function isVisibleUpToRoot(node, root) {
+  let child = node;
+  while (child) {
+    if (child === root) return true;
+    if (!child.visible) return false;
+    child = child.parent;
+  }
+  console.error(`Root ${root} is not in the hierarchy of node ${node}`);
+  return false;
+}
+
+export const computeLocalBoundingBox = (function() {
+  const vertex = new THREE.Vector3();
+  const rootInverse = new THREE.Matrix4();
+  const toRootSpace = new THREE.Matrix4();
+  return function computeLocalBoundingBox(root, box, excludeInvisible) {
+    box.makeEmpty();
+    root.updateMatrices();
+    rootInverse.getInverse(root.matrixWorld);
+    root.traverse(node => {
+      if (excludeInvisible && !isVisibleUpToRoot(node, root)) {
```

Comment:
Hmm slightly scary here that we are calling `isVisibleUpToRoot` here on every node in the subtree, lots of duplicate work happening.. Might make sense to do the traverse manually and roll in the visibility check to cull out duplicating work. That or we come up with a similar solution to matrixNeedsUpdate but for visibility, which sounds painful.

Reply:
You're right, but since this is called very infrequently I'm not very concerned. (Once per mesh, and once per opened menu each time a camera is toggled, but the menus do not have deep hierarchies.)